src/main/java/com/spotify/dns/ChangeNotifier.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2012-2015 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.spotify.dns;
+
+import java.util.Set;
+
+public interface ChangeNotifier<T> {
+
+  /**
+   * Get the current set of endpoints.
```

Comment:
Should be current set of 'records' or something, I think, rather than 'endpoints'?